### PR TITLE
Rewrite to unix domain socket. Leave Tkinter as an option for debugging

### PR DESCRIPTION
**Pull request by JanSvoboda**
Hi, I have updated python implementation to use unix domain sockets for communication and the script now runs without tkinter, which I have left for debugging purposes.

### CODE REVIEW

**Comment by varjolintu** [comment]
Thank you for this! I will merge the PR after some testing if no issues are found.

**Comment by varjolintu** [comment]
This doesn't work with macOS without Tkinter. Did you test it by yourself?

**Comment by JanSvoboda** [comment]
No. Unfortunatelly not, because I don't have any macOS to test it on. I don't have any at my disposal right now. I did test it on Windows and Linux

Odesláno z iPhonu

1. 4. 2018 v 18:37, Sami Vänttinen <notifications@github.com>:

**Comment by JanSvoboda** [comment]
I have changed shebang to use python2.7 for macOS. I did test macOS (High Sierra) and it worked fine for me - both options - with and without tkinter.

File: python_version/keepassxc-proxy [hunk 1]
```diff
@@ -12,108 +12,142 @@ import threading
...
+                self.quit()
+                return
+            self.log("Received %s" % message)
+
+            # Handle requests from keepassxc-browser.
+            messageJSON = json.loads(message)
+            if messageJSON['action']:
+                sock.send(message)
+                try:
+                    resp, server = sock.recvfrom(4096)
```
Comment (varjolintu) [inline python_version/keepassxc-proxy]: In the next version the maximum data size is 1024*1024. Please change this hard-coded value to a const variable.

**Comment by varjolintu** [comment]
Working here too. Nice work! Also see the following PR https://github.com/keepassxreboot/keepassxc/pull/1811 for some changes in the future.

**Comment by JanSvoboda** [comment]
I have prepared python version for issue 1811. Buffer size is separated as const variable

**Comment by varjolintu** [comment]
Sorry for such a long delay. I'll merge this PR soon if I don't find any more issues.

**Pull request merged**